### DISCUSSION TITLE
Error using phaseList/checkSinglePhase (GND calculations)

### DISCUSSION THREAD

**Discussion by jucego333** [mtex-toolbox/mtex#741] [Ask Anything] [answered]
Hello everyone, I am new using Mtex and really interested to get familiar with this. I want to use the software to determine the GND of EBSD data containing 4 phases. However, when I follow the documentation from https://mtex-toolbox.github.io/GND.html, I got the following error at matlab:

"Error using phaseList/checkSinglePhase (line 307)

----------------------------------------------------------------
Your variable contains the phases: Iron fcc, Iron bcc (old)

However, your are executing a command that is only permitted for a single phase!

Please read the chaper "select EBSD data" for how to restrict EBSD data or grains
to a single phase."

I know that for GND, it is important to consider all the phases of the sample. Can anybody suggest to me how to deal with this?

By the  way I am using MTEXT 5.6.0

Best regards.
```

[text1.txt](https://github.com/mtex-toolbox/mtex/files/5916955/text1.txt)

**Comment by ralfHielscher** [comment, marked as answer]
You need to compute the GND for each phase separately. The reason is that each phase has different dislocation system, constants,... You can later combine the results into one map if you want to do so.

**Reply by jucego333** [reply]
Thank you I will consider this.
